### DISCUSSION TITLE
bug: Potential bug with AzureAD auth setup

### DISCUSSION THREAD

**Discussion by mprudra** [langfuse/langfuse#1466] [Support]
I'm constantly facing trouble with my Langfuse instance with AzureAD auth flow. While same instance is working for rest of my teammates. Tried clearing cookies, using different browser, still the same issue.

I'm using Kubernetes to host the instance (latest build of Langfuse).

From logs:
```
[next-auth][error][OAUTH_CALLBACK_ERROR] 
https://next-auth.js.org/errors#oauth_callback_error invalid_grant (AADSTS54005: OAuth2 Authorization code was already redeemed, please retry with a new valid code or use an existing refresh token. Trace ID: c2f453f9-3b83-400e-85cf-7a1d3c385200 Correlation ID: 6ab3e2f4-e65a-4241-a974-b280b497dfea Timestamp: 2024-03-22 07:05:37Z) {
  error: OPError: invalid_grant (AADSTS54005: OAuth2 Authorization code was already redeemed, please retry with a new valid code or use an existing refresh token. Trace ID: c2f453f9-3b83-400e-85cf-7a1d3c385200 Correlation ID: 6ab3e2f4-e65a-4241-a974-b280b497dfea Timestamp: 2024-03-22 07:05:37Z)
      at processResponse (/app/node_modules/openid-client/lib/helpers/process_response.js:38:13)
      at Client.grant (/app/node_modules/openid-client/lib/client.js:1354:22)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async Client.callback (/app/node_modules/openid-client/lib/client.js:493:24)
      at async oAuthCallback (/app/node_modules/next-auth/core/lib/oauth/callback.js:109:16)
      at async Object.callback (/app/node_modules/next-auth/core/routes/callback.js:52:11)
      at async AuthHandler (/app/node_modules/next-auth/core/index.js:208:28)
      at async NextAuthApiHandler (/app/node_modules/next-auth/next/index.js:22:19)
      at async NextAuth._args$ (/app/node_modules/next-auth/next/index.js:108:14)
      at async /app/node_modules/@sentry/nextjs/cjs/common/wrapApiHandlerWithSentry.js:136:41 {
    name: 'OAuthCallbackError',
    code: undefined
  },
  providerId: 'azure-ad',
  message: 'invalid_grant (AADSTS54005: OAuth2 Authorization code was already redeemed, please retry with a new valid code or use an existing refresh token. Trace ID: c2f453f9-3b83-400e-85cf-7a1d3c385200 Correlation ID: 6ab3e2f4-e65a-4241-a974-b280b497dfea Timestamp: 2024-03-22 07:05:37Z)'
```

**Comment by marcklingen** [comment]
This is interesting, I unfortunately cannot reproduce it. I'd try deleting the user in the `Accounts` table to reset the user oauth session

**Reply by mprudra** [reply]
Didn't work, I had even tried deleting everything, a fresh setup, didn't work either.

**Reply by marcklingen** [reply]
Sorry, can't help debug this without additional information. We use plain AzureAD via Auth.js and this runs in prod for lots of customers. Feel free to share more details in case you want to figure out together how to fix this in your instance

**Reply by m-a-zaman** [reply]
Hi, do you remember how have you resolved this issue?

**Comment by rsam444** [comment]
@mprudra  @m-a-zaman  I had a similar issue and i was able to resolve it by adding the below annotation in my nginx ingress object for langfuse.

**nginx.ingress.kubernetes.io/proxy-buffer-size: 8k**

If your backend sends large response headers (e.g., big cookies, lots of custom headers, or large JWT tokens in headers), and the buffer is too small, NGINX may return a 502 Bad Gateway or 504 error.

Hope this helps.
